cpp/src/arrow/compute/kernels/scalar_set_lookup.cc

```diff
@@ -274,10 +280,12 @@ struct IndexInVisitor {
     const auto& state = checked_cast<const SetLookupState<NullType>&>(*ctx->state());
```

Comment:
Would you mind modifying this visit overload to:
```diff
-  Status Visit(const DataType& type) {
-    DCHECK_EQ(type.id(), Type::NA);
+  Status Visit(const DataType& type) { DCHECK(false) << "IndexIn " << type; }
+  Status Visit(const NullType&) {
```

Reply:
Sure, I have modified to that. Besides, for `Status Visit(const DataType& type)`, I add a return `Status::NotImplemented("IndexIn has no implementation with value type ", type)`